### PR TITLE
Default to always testing in async & sync

### PR DESCRIPTION
**Pull request by LukeButters**
# Background

 [SC-51287]

Some tests are not yet converted to async and sync, since either they don't work when converted or is enough work it probably makes sense to be done in another PR.

# How to review this PR

Quality :heavy_check_mark:

# Pre-requisites

- [ ] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/master/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [ ] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [ ] I have considered appropriate testing for my change.

### CODE REVIEW

File: source/Halibut.Tests/ParallelRequestsFixture.cs [hunk 1]
```diff
@@ -50,7 +50,7 @@ public async Task MultipleRequestsCanBeInFlightInParallel(ClientAndServiceTestCa
                     {
                         var requestStartedPath = $"{requestStartedFilePathBase}-{iteration}";
                         requestStartedFilePaths.Add(requestStartedPath);
-                        lockService.WaitForFileToBeDeleted(lockFile, requestStartedPath);
+                        lockService.WaitForFileToBeDeletedAsync(lockFile, requestStartedPath).GetAwaiter().GetResult();
```
Comment (nathanwoctopusdeploy) [inline source/Halibut.Tests/ParallelRequestsFixture.cs:53]: nit await

File: source/Halibut.Tests/ProxyFixture.cs [hunk 2]
```diff
@@ -44,6 +49,9 @@ public async Task OctopusCanNotSendMessagesToTentacle_WithEchoService_AndABroken
             {
                 await clientAndService.HttpProxy!.StopAsync(CancellationToken.None);
 
+                // var echo = clientAndService.CreateClient<IEchoService, IAsyncClientEchoService>();
```
Comment (nathanwoctopusdeploy) [inline source/Halibut.Tests/ProxyFixture.cs:52]: ??

File: source/Halibut.Tests/TestServices/Async/IAsyncClientReturnSomeDataStreamService.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+using System;
+using System.Threading.Tasks;
+
+namespace Halibut.Tests.TestServices.Async
+{
+    public interface IAsyncClientReturnSomeDataStreamService
+    {
+        public Task<DataStream> SomeDataStreamAsync();
```
Comment (nathanwoctopusdeploy) [inline source/Halibut.Tests/TestServices/Async/IAsyncClientReturnSomeDataStreamService.cs:8]: nit: remove public

**Review by nathanwoctopusdeploy — APPROVED** [review]
LGTM

File: source/Halibut.Tests/ProxyFixture.cs (hunk 2 above)
Comment (LukeButters) [reply source/Halibut.Tests/ProxyFixture.cs:52]: This is the async version of the test, which doesn't work. Will remove

**Pull request merged**